packages/react-lowcode/src/codegen/list.ts

```diff
@@ -17,32 +18,81 @@ export function isSelectedDataTable(sourceCode:string, tablePosition: SourceLine
 export async function addColumn(typesSourceCode: string, 
     io: CodeRW, 
     sourceCode:SourceLineCol, 
-    options: InsertOptions): Promise<string | undefined>{
+    options: InsertOptions, 
+    code?: string): Promise<string | undefined>{
         
     const property: Property = getEntityProperty(typesSourceCode, options.property, options.entityName)[0]
     let generatedSource = undefined
+    let isGrommet = false
+    if (code)
+        isGrommet = isGrommetRepository(code)
 
     if(property){
-        generatedSource = await insertColumn(sourceCode, 
-        {entityField: property, index: options.index}, 
-        io)
+        if (!isGrommet) {
+            generatedSource = await insertColumn(
+                sourceCode, 
+                {
+                    entityField: property, index: options.index
+                }, 
+                io
+            )
+        } else {
+            const myClassFile = parseGraphqlTypes(typesSourceCode)
+            const entity = sourceFileEntity(myClassFile, options.entityName)
+    
+            let ent : any = {};
+            ent.properties = [property]
+            ent.getName = entity?.getName
+            options.index = (options.index as number) + 1
+            generatedSource = await insertColumnToDataTableGrommet(
+                sourceCode,
+                {
+                    entityField: property,
+                    index: options.index,
+                    entity: ent
+                },
+                io
+            )
+        }    
     }
 
     return generatedSource
 }
 
 export async function deleteColumn(io: CodeRW, 
     sourceCode:SourceLineCol, 
-    options: DeleteOptions): Promise<string | undefined> {
+    options: DeleteOptions, 
+    code?: string): Promise<string | undefined> {
+    let isGrommet = false
+    if(code) 
+        isGrommet = isGrommetRepository(code)
 
-let generatedSource = await fDeleteColumn(sourceCode, options, io);
+    if (isGrommet) 
+        options.index = options.index + 1
```

Comment:
why do we need this strange thing?

Reply:
we need it because indexing in grommet is different as it is in material-ui and i thought it is better to just check if its grommet or not

Reply:
can you explain that on an example please? It is a antipattern to modify input values. 
At least, copy the options before incrementing instance while propagating options to consequent functions calls.
Same applies to the code below.

Reply:
in material-ui elements indexing is starting with `0,1,2,.....`. I found that in grommet indexing is probably starting `1,2,3,4`.